documents/anchor_task.go

```diff
@@ -116,16 +120,24 @@ func (d *documentAnchorTask) RunTask() (res interface{}, err error) {
 }
 
 // InitDocumentAnchorTask enqueues a new document anchor task and returns the txID.
-func InitDocumentAnchorTask(tq queue.TaskQueuer, txService transactions.Service, accountID identity.CentID, modelID []byte) (uuid.UUID, error) {
-	tx, err := txService.CreateTransaction(accountID, documentAnchorTaskName)
+// TODO [TXManager] migrate this to use TxManager
+func InitDocumentAnchorTask(tq queue.TaskQueuer, txService transactions.Manager, accountID identity.CentID, modelID []byte, txID uuid.UUID) (uuid.UUID, error) {
+	var tx *transactions.Transaction
+	var err error
+	if txID != uuid.Nil {
+		tx, err = txService.GetTransaction(accountID, txID)
```

Comment:
As this is an Init function, in which scenario do we expect a txID to be passed and exists before hand?

Reply:
This is a possibility right now because of the some transactions are implemented but would be removed later